### PR TITLE
Ember upgrade to 2.18.2

### LINKED ISSUE #1365: Update ember and ember-data to the latest versions (goal is at least v2@latest)
Let's move our ember and be prepared for the v3, which is in beta now. The latest version is 2.18 at the moment of writing. Whoever picks it please use the [deprecation guide](https://www.emberjs.com/deprecations/) and get rid of all the deprecation messages occured

### PR DESCRIPTION
**Pull request by stukalin**
Fixes #1365.

**Changes proposed in this pull request:**
- ember-source/-data/-cli were upgraded to 2.18.2
- blueprints were regenerated
- automigrate for rfc 176 has been run
- package-lock.json has been removed

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: app/controllers/appointments.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Controller from '@ember/controller';
```
Comment (stukalin) [inline app/controllers/appointments.js]: These empty controllers were created as workarounds: there was a failure with generated controllers.

File: config/deprecation-workflow.js [hunk 2]
```diff
@@ -3,5 +3,6 @@ window.deprecationWorkflow.config = {
   workflow: [
     { handler: 'silence', matchId: 'ember-getowner-polyfill.import' },
     { handler: 'silence', matchId: 'ember-views.did-init-attrs' },
+    { handler: 'silence', matchId: 'ember-inflector.globals' },
```
Comment (stukalin) [inline config/deprecation-workflow.js]: This should be removed.

**Comment by tangollama** [comment]
@stukalin the CI seems to be failing on the start. I'm assuming this is a WIP. Regardless, thanks for beginning to slay this beast.

**Comment by stukalin** [comment]
@tangollama not anymore:)

**Pull request merged**